src/librustc_driver/driver.rs

```diff
@@ -662,6 +647,19 @@ pub fn phase_2_configure_and_expand(sess: &Session,
         krate
     })?;
 
+    // Needs to go *after* expansion to be able to check the results
+    // of macro expansion.
+    time(time_passes, "complete gated feature checking 1", || {
+        sess.track_errors(|| {
+            let features = syntax::feature_gate::check_crate(sess.codemap(),
+                                                             &sess.parse_sess.span_diagnostic,
+                                                             &krate,
+                                                             &attributes,
+                                                             sess.opts.unstable_features);
+            *sess.features.borrow_mut() = features;
+        })
+    })?;
+
```

Comment:
There is already a second feature gate checking pass just a bit later in the same function (“complete gated feature checking 2”). I'm not sure this first pass is necessary at tall. I considered removing it as part of #32791 but then thought I'd do a separate PR.